ambry-rest/src/main/java/com.github.ambry.rest/NettyServer.java

```diff
@@ -49,27 +49,21 @@
   private final NettyMetrics nettyMetrics;
   private final EventLoopGroup bossGroup;
   private final EventLoopGroup workerGroup;
-  private final RestRequestHandler requestHandler;
-  private final PublicAccessLogger publicAccessLogger;
-  private final RestServerState restServerState;
+  private final LinkedHashMap<String, ChannelHandler> channelHandlerInfoList;
   private final Logger logger = LoggerFactory.getLogger(getClass());
 
   /**
    * Creates a new instance of NettyServer.
    * @param nettyConfig the {@link NettyConfig} instance that defines the configuration parameters for the NettyServer.
    * @param nettyMetrics the {@link NettyMetrics} instance to use to record metrics.
-   * @param requestHandler the {@link RestRequestHandler} that can be used to submit requests that need to be handled.
-   * @param publicAccessLogger the {@link PublicAccessLogger} that can be used for public access logging
-   * @param restServerState the {@link RestServerState} that can be used to check the health of the system
-   *                              to respond to health check requests
+   * @param channelHandlerInfoList Linked list of pairs of {@link ChannelHandler} name and {@link ChannelHandler}s to
```

Comment:
An ordered list of pairs of {@link ChannelHandler} name and {@link ChannelHandler}s...